conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,252 @@
+Test-Case: output
+Description: Test trap-expr when the expression is a nil-literal.
+Labels: trap-expr, error-type, null, nil-literal, is-expr
+
+function init() {
+    ()|error t1 = trap ();
+    io:println(t1 is ()); // @output true
+
+    error? t2 = trap null;
+    io:println(t2 is ()); // @output true
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a boolean-literal.
+Labels: trap-expr, error-type, boolean, boolean-literal
+
+function init() {
+    boolean|error t = trap false;
+    io:println(t); // @output false
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a numeric-literal.
+Labels: trap-expr, error-type, int, float, decimal
+
+function init() {
+    int|error t1 = trap 2;
+    io:println(t1); // @output 2
+    float|error t2 = trap 2e3f;
+    io:println(t2); // @output 2000.0
+    decimal|error t3 = trap 2e355d;
+    io:println(t3); // @output 2E+355
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a string-literal.
+Labels: trap-expr, string
+
+function init() {
+    string|error t1 = trap "Hello";
+    io:println(t1); // @output Hello
+    t1 = trap "Hello\tWorld";
+    io:println(t1); // @output Hello	World
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a byte-array-literal.
+Labels: trap-expr, error-type, byte, byte-array-literal
+
+function init() {
+    byte[]|error t1 = trap base16 `aeeecdefabcd12345567888822`;
+    io:println(t1); // @output [174,238,205,239,171,205,18,52,85,103,136,136,34]
+    t1 = base64 `aGVsbG8gYmFsbGVyaW5hICEhIQ==`;
+    io:println(t1); // @output [104,101,108,108,111,32,98,97,108,108,101,114,105,110,97,32,33,33,33]
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a string-template-expr.
+Labels: trap-expr, error-type, string-template-expr
+
+function init() {
+    string|error t1 = trap string `Hello World!`;
+    io:println(t1); // @output Hello World!
+    string s = string `Hello World!`;
+    string|error t2 = string `${s}`;
+    io:println(t2); // @output Hello World!
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a xml-template-expr.
+Labels: trap-expr, error-type, xml-template-expr
+
+function init() {
+    xml|error t1 = trap xml `<Greeting>Hello!</Greeting>`;
+    io:println(t1); // @output <Greeting>Hello!</Greeting>
+}
+
+Test-Case: output
+Description: Test trap-expr when the expression is a list-constructor-expr.
```

Comment:
Since there are multiple tests for `list-constructor-expr`, shall we add a test number in here?
```suggestion
Description: Test trap-expr when the expression is a list-constructor-expr. Test 1.
```